### PR TITLE
Delete old diagnostics

### PR DESCRIPTION
**Pull request by MaxThevenet**
This PR aims at deleting the old implementation of diagnostics. It contains:
- Some reorganization of the diagnostics section of the documentation
- Updates to use the new diagnostics
- Deletion the previous implementation for standard diagnostics and OpenPMD diagnostics
- Changes in CI scripts to use the new diagnostics

While iterating to have this PR working, two minor fixes are included in this PR:
- in `MultiDiagnostics::FilterComputePackFlush`, a `return` statement was replaced by a `continue` statement.
- `diagnostics.ndiags` is not needed anymore, the number of diagnostics are simply ready from the number of elements in `diagnostics.diags_names`.

Note that this is far from totally replacing the previous implementation of diags, here's what remains to be done:
- reformat back-transformed diags to use the new framework
- in-situ diags

### CODE REVIEW

**Comment by dpgrote** [comment]
The picmi interface will need to be updated since it has code setting up the old diagnostics. Were you planning on doing that? I can volunteer to make the changes in the separate PR.

**Comment by MaxThevenet** [comment]
I wasn't sure how to do that, so please go ahead and update it. Note that currently these changes only affect field diagnostics, but we plan on expanding this to particles soon, which will require further changes.

Thanks a lot for your help!

File: Regression/WarpX-tests.ini [hunk 1]
```diff
@@ -670,7 +671,7 @@ particle_tolerance = 1.e-12
 [Python_Langmuir]
 buildDir = .
 inputFile = Examples/Tests/Langmuir/PICMI_inputs_langmuir_rt.py
-runtime_params =
+runtime_params = diag1.file_prefix=Python_Langmuir_plt
```
Comment (dpgrote) [inline Regression/WarpX-tests.ini:674]: Note that runtime_params will be ignored for the Python test cases. In my PR, I'll make the appropriate changes to the input files, for example so the data is written out to the plot files with the same names as before.

Comment (MaxThevenet) [reply Regression/WarpX-tests.ini:674]: Good point, thanks, then I will delete these lines as they are useless.

**Comment by atmyers** [comment]
Maybe you could have a list of specific entries to ignore when checking for unused ParmParse variables.

**Comment by MaxThevenet** [comment]
I think this PR is ready and the tests pass \o/. @atmyers do you want to have a look, or should we merge it?

**Comment by MaxThevenet** [comment]
Ah yep, thanks the the label @ax3l.

**Pull request merged**